### PR TITLE
doc: Update Destination Cache Docs

### PR DESCRIPTION
**Pull request by marikaner**
Update documentation on caching and isolation, as we removed the enums. I generally simplified this documentation a bit and added some details on questions that arose for me while reading.

Closes https://github.com/SAP/cloud-sdk-backlog/issues/1014

### CODE REVIEW

File: docs-js/features/connectivity/destination-cache-isolation.mdx [hunk 1]
```diff
@@ -4,141 +4,204 @@ title: Destination Cache and Isolation
...
-The isolation strategy is automatically detected and applied based on the given JWT:
-
-- No JWT: Isolation is set to `Tenant` and value for the tenant is the provider account.
-- JWT without `user_id`: Isolation is set to `Tenant` and the value for the tenant is the `zid` of the JWT.
-- JWT with `user_id`: Isolation is set to `TenantUser` and values are taken from `zid` and `user_id`.
+## Registered Destination Cache
 
-You can override the automatically detected isolation strategy.
+When [registering destinations](./destination.mdx#register-destination) using the [`registerDestination()`](pathname:///api/v3/functions/sap_cloud_sdk_connectivity.registerDestination.html) function, destinations are stored in a separate cache.
+This cache is always enabled and be behaves similarly to the destination cache discussed above.
```
Comment (FrankEssenberger) [inline docs-js/features/connectivity/destination-cache-isolation.mdx:183]:
```suggestion
This cache is always enabled and behaves similarly to the destination cache discussed above.
```

File: docs-js/features/connectivity/destination-cache-isolation.mdx [hunk 2]
```diff
@@ -4,141 +4,204 @@ title: Destination Cache and Isolation
...
+It uses the same default [isolation strategies](#isolation-strategy).
```
Comment (FrankEssenberger) [inline docs-js/features/connectivity/destination-cache-isolation.mdx:184]: [pp] With 'same' you mean the one explained in the section `### Isolation Strategy` right? If so I would link it to make clear which one.

File: docs-js/features/connectivity/destination-cache-isolation.mdx [hunk 3]
```diff
@@ -4,141 +4,204 @@ title: Destination Cache and Isolation
...
+
+```ts
+import { registerDestination } from '@sap-cloud-sdk/connectivity';
+
+const destination = {
+  name: 'DESTINATION',
+  url: 'https://mys4hana.com'
+};
+
+registerDestination(destination, { jwt: 'yourJWT' });
```
Comment (FrankEssenberger) [inline docs-js/features/connectivity/destination-cache-isolation.mdx:204]: [req] I think you used a different string for jwt value above. Align them all.

**Review by FrankEssenberger — APPROVED** [review]
LGTM - I left a few request and comments nothing big.

File: docs-js/features/connectivity/destination-cache-isolation.mdx (hunk 2 above)
Comment (marikaner) [reply docs-js/features/connectivity/destination-cache-isolation.mdx:184]: I tried to make it more clear, I think the second half of the sentence was basically missing.

**Pull request merged**